src/ansys/sound/core/sound_composer/source_control_time.py

```diff
@@ -0,0 +1,142 @@
+
+
+class SourceControlTime(SourceControlParent):
+    """Sound Composer's time source control class.
+
+    This class stores the source control, that is the control parameter values over time, used by
+    the Sound Composer to generate a sound from a BBN or harmonic source.
+    """
+
+    def __init__(self, file_str: str = ""):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        file_str : str, default: ""
+            Path to the control data file (WAV format or text format with header
+            `AnsysSound_SoundSamples`).
+        """
+        super().__init__()
+
+        # Define DPF Sound operators.
+        self.__operator_load = Operator(ID_LOAD_FROM_TEXT)
+
+        if len(file_str) > 0:
+            if file_str.endswith(".wav"):
+                self.load_from_wave_file(file_str)
+            else:
+                self.load_from_text_file(file_str)
+        else:
+            self.control = None
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if self.control is None:
+            return "Not set"
+        else:
+            support_data = self.control.time_freq_support.time_frequencies.data
+            str_duration = f"{support_data[-1]:.1f} s" if len(support_data) > 0 else "N/A"
+            return (
+                f"Unit: {self.control.unit}\n"
+                f"Duration: {str_duration}\n"
+                f"Min - max: {self.control.data.min():.1f} - {self.control.data.max():.1f} "
+                f"{self.control.unit}"
+            )
+
+    @property
+    def control(self) -> Field:
+        """Control profile (control parameter values over time)."""
+        return self.__control
+
+    @control.setter
+    def control(self, control: Field):
+        """Set the control."""
+        if control is not None:
+            if not (isinstance(control, Field) or control is None):
+                raise PyAnsysSoundException(
+                    "Specified control profile must be provided as a DPF field."
+                )
+
+            if len(control.data) == 0:
+                raise PyAnsysSoundException(
+                    "Specified control profile must have at least one element."
+                )
+
+        self.__control = control
+
+    def load_from_wave_file(self, file_str: str):
+        """Load control data from a WAV file.
+
+        Parameters
+        ----------
+        file_str : str
+            Path to the WAV file.
+        """
+        loader = LoadWav(file_str)
+        loader.process()
+        self.control = loader.get_output()[0]
+
+    def load_from_text_file(self, file_str: str):
+        """Load control data from a text file.
+
+        Parameters
+        ----------
+        file_str : str
+            Path to the text file. The text file shall have the same text format (with the header
+            `AnsysSound_SoundSamples`), as supported by Ansys Sound SAS.
```

Comment:
@ansaminard it's not AnsysSound_**SoundSamples** it's, **AnsysSound_BBN**

Reply:
@a-bouth Well from what I can see, only AnsysSound_SoundSamples is supported in operator load_sound_samples_from_txt (at least this format works, while I get an error if I use an AnsysSound_BBN file - whether v1 or v2 -)
Regardless, from what I understand, the BBN format contains spectrum data over various control values, while this class here, according to specifications, is supposed to only contain the control parameter values over time...